frontend/public/components/secrets/create-secret.tsx

```diff
@@ -139,6 +160,211 @@ const withSecretForm = (SubForm) => class SecretFormComponent extends React.Comp
+        loadedData.username = _.first(auth);
+        loadedData.password = _.last(auth);
+      }
+    }
+
+    this.state = {
+      isNewFormat: _.isEmpty(loadedData) || Object.keys(this.props.stringData)[0] === AUTHS_SYMBOL,
+      address: _.get(loadedData, 'address') || '',
+      username: _.get(loadedData, 'username') || '',
+      password: _.get(loadedData, 'password') || '',
+      email: _.get(loadedData, 'email') || '',
+    };
+    this.changeData = this.changeData.bind(this);
+  }
+  changeData(event) {
+    this.setState({
+      [event.target.name]: event.target.value
+    } as CreateConfigSubformState, () => this.constructObj());
+  }
+  constructObj() {
+    if (this.state.address && this.state.username && this.state.password && this.state.email) {
+      const secretData = {
+        [this.state.address]: {
+          auth: window.btoa(`${this.state.username}:${this.state.password}`),
+          username: this.state.username,
+          password: this.state.password,
+          email: this.state.email
+        }
+      };
+      this.props.onChange(this.state.isNewFormat ? {[AUTHS_SYMBOL]: secretData} : secretData);
+    }
+  }
+  render() {
+    return <React.Fragment>
+      <div className="form-group">
+        <label className="control-label">Registry Server Address</label>
+        <div>
+          <input className="form-control"
+            id="address"
+            type="text"
+            name="address"
+            onChange={this.changeData}
+            value={this.state.address}
+            required />
+        </div>
+      </div>
+      <div className="form-group">
+        <label className="control-label">Username</label>
+        <div>
+          <input className="form-control"
+            id="username"
+            type="text"
+            name="username"
+            onChange={this.changeData}
+            value={this.state.username}
+            required />
+        </div>
+      </div>
+      <div className="form-group">
+        <label className="control-label">Password</label>
+        <div>
+          <input className="form-control"
+            id="password"
+            type="password"
+            name="password"
+            onChange={this.changeData}
+            value={this.state.password}
+            required />
+        </div>
+      </div>
+      <div className="form-group">
+        <label className="control-label">Email</label>
+        <div>
+          <input className="form-control"
+            id="email"
+            type="text"
+            name="email"
+            onChange={this.changeData}
+            value={this.state.email}
+            required />
+        </div>
+      </div>
+    </React.Fragment>;
+  }
+}
+
+class UploadConfigSubform extends React.Component<UploadConfigSubformProps, UploadConfigSubformState> {
+  constructor (props) {
+    super(props);
+    this.state = {
+      configFile: _.isEmpty(this.props.stringData) ? '' : JSON.stringify(this.props.stringData),
```

Comment:
I'm confused here... Isn't the config file one value of one key? Why is it set to all of stringData?